src/AppInstallerCLICore/Workflows/ConfigurationFlow.cpp

```diff
@@ -959,6 +961,87 @@ namespace AppInstaller::CLI::Workflow
+            {
+                return Settings::ConfigurationAllowedZonesOptions::UntrustedSites;
+            }
+
+            return static_cast<Settings::ConfigurationAllowedZonesOptions>(dwZone);
+        }
+
+        // Validate group policy for a given zone.
+        bool IsBlockedByGroupPolicy(Execution::Context& context, const Settings::ConfigurationAllowedZonesOptions zone)
+        {
+            auto configurationPolicies = Settings::GroupPolicies().GetValue<Settings::ValuePolicy::ConfigurationAllowedZones>();
+            if (!configurationPolicies.has_value())
+            {
+                AICLI_LOG(Config, Warning, << "ConfigurationAllowedZones policy is not set");
+                return false;
+            }
+
+            if (configurationPolicies->find(zone) == configurationPolicies->end())
+            {
+                AICLI_LOG(Config, Warning, << "Configuration is not configured in the zone " << zone);
+                return false;
+            }
+
+            auto isAllowed = configurationPolicies->at(zone);
+            if(!isAllowed)
+            {
+                context.Reporter.Error() << "Configuration is disabled for Zone: " << zone << std::endl;
+                return true;
+            }
+
+            AICLI_LOG(Config, Info, << "Configuration is configured in zone " << zone << " with value " << (isAllowed ? "allowed" : "blocked"));
+            return false;
+        }
+
+        // Validate smart screen for a given url.
+        bool IsBlockedBySmartScreen(Execution::Context& context, const std::string& url)
+        {
+            auto response = AppInstaller::UriValidation::UriValidation(url);
+            switch (response.Decision())
+            {
+            case AppInstaller::UriValidation::UriValidationDecision::Block:
+                context.Reporter.Error() << std::endl << "Blocked by smart screen" << std::endl << "Feedback: " << response.Feedback() << std::endl;
+                return true;
+            case AppInstaller::UriValidation::UriValidationDecision::Allow:
+            default:
+                return false;
+            }
+        }
+
+        bool IsSmartScreenRequired(Settings::ConfigurationAllowedZonesOptions zone)
+        {
+            return zone == Settings::ConfigurationAllowedZonesOptions::Internet
+                || zone == Settings::ConfigurationAllowedZonesOptions::UntrustedSites;
+        }
+
+        // Evaluate a given uri for configuration.
+        HRESULT EvaluateUri(Execution::Context& context, const std::string& uri)
+        {
+            auto zone = GetUriZone(uri);
+            if(IsBlockedByGroupPolicy(context, zone))
+            {
+                return APPINSTALLER_CLI_ERROR_BLOCKED_BY_POLICY;
+            }
+
+            if (IsSmartScreenRequired(zone) && IsBlockedBySmartScreen(context, uri))
+            {
+                return APPINSTALLER_CLI_ERROR_SOURCE_NOT_SECURE;
+            }
+
+            return NO_ERROR;

```

Comment:
```suggestion
            return S_OK;
```
No functional difference, but `NO_ERROR` is a Win32 error code definition while `S_OK` is an HRESULT definition.